### DISCUSSION TITLE
Generate 3D objects for species block files

### DISCUSSION THREAD

**Discussion by Keng-Lee** [epochpic/epoch#737] [Q&A]
Hello, I have successfully created a 2D .dat file and have allowed EPOCH to read it without any issues, as outlined in the user manual. I would like to inquire whether the 3D component can also be accomplished using .dat files.

**Comment by Status-Mirror** [comment]
Hi @Keng-Lee,

I'm not sure what you mean by `.dat` file. Where are you reading this in the documentation?

You can define 3D species densities in the same way as 2D, but you also have extra variables like $z$ and $dz$ you can use with the [maths parser](https://epochpic.github.io/documentation/code_details/maths_parser.html).

Hope this helps,
Stuart

**Reply by Keng-Lee** [reply]
Thank you for your reply. I noticed it in the B Simple Binary Files section, which describes the two-dimensional method but does not mention the three-dimensional method.
![螢幕擷取畫面 2024-11-12 213817](https://github.com/user-attachments/assets/fca9fdcd-c823-4333-a111-0e170634851c)
